R/runEmployment.R

```diff
@@ -0,0 +1,304 @@
+             "Tech|Electricity|Storage|Battery|For PV|OM Cost|fixed", 
+             NULL
+    )
+    # only capital costs
+    report_tech <- input_mif_mp[,,var]
+    cap_costs <- report_tech[getRegions(x), seq(2015, 2050, 5), "Capital Costs",pmatch=TRUE]# selecting years until 2050 and removing "World" region
+
+    cap_costs <- cap_costs[, , ] / setYears(cap_costs[, "y2020", ], NULL) # costs relative to 2020
+    
+    # Evolution of employment factor (CI & Manf) with time depends on capital costs
+    # the job intensity decreases with capital costs for techs, relative to 2015
+    var_in_x <- getItems(x, dim = 3.1) # variables from employment factor magpie object
+    var_in_rem <- c("Coal", "Gas", "Nuclear", "Biomass", "Hydro", "Hydro", "Wind|Onshore", "Wind|Offshore", "Solar|PV",
+                    "Geothermal", "Solar|CSP", "Oil", "Solar|PV", "Storage") # variables from remind (order matters)
+    var_comb <- paste(var_in_x, var_in_rem, sep = ".") # concatenating the two for looping to work
+    
+    for (i in var_comb) {
+      x[getRegions(x), , sub("\\..*", "", i)][, , c("CI", "Manf")] <- x[getRegions(x), , sub("\\..*", "", i)][, , c("CI", "Manf")] * setNames(cap_costs[getRegions(x), getYears(x), sub(".*\\.", "", i), pmatch = TRUE], NULL)
+    }
+    # OM fixed costs used to calculate  decline factors for O&M employment factors
+    fixed_costs <- report_tech[getRegions(x), seq(2015, 2050, 5), "fixed",pmatch=TRUE]# selecting years until 2050 and removing "World" region
+
+    fixed_costs <- fixed_costs[, , ] / setYears(fixed_costs[, "y2020", ], NULL) # costs relative to 2020
+    for (i in var_comb) {
+      x[getRegions(x), , sub("\\..*", "", i)][, , "OM"] <- x[getRegions(x), , sub("\\..*", "", i)][, , "OM"] * setNames(fixed_costs[getRegions(x), getYears(x), sub(".*\\.", "", i), pmatch = TRUE], NULL)
+    }
+    
+    # decline factors for coal fuel supply until 2050 are projected depending on
+    # historical patterns and convergence between some countries.
+    # See calcLabourProductivity for more information.
+    coal_ef <- calcOutput("CoalLabourProductivity", subtype = "Employment_factor")
+    coal_ef <- coal_ef / setYears(coal_ef[, 2020, ], NULL)
+    coal_ef[which(coal_ef == "NaN")] <- 0
+    x[, getYears(x), "Coal.Fuel_supply"] <- x[, getYears(x), "Coal.Fuel_supply"] * setNames(coal_ef[, getYears(x), ], NULL)
+  }
+  # if Employment factors don't change
+  if (decline == "static") {
+    # do nothing, as employment factors have already been read
+  }
+  
+  ## Other external parameters--------------------------
+
+  # share of rooftop in total spv, wind offshore in total wind, and hydro small in total hydro
+  share <- calcOutput("DspvShare")
+  
+  ## ----------------------------------------------
+  
+  ## New and Absolute capacity from REMIND---------------------------------
+  # filtering required capacity variables i.e., only new and existing coal capacities
+#  remind_output <- reportCapacity(gdx)
```

Comment:
Can you explain in a comment why these lines are commented out and when/under which circumstances they should be commented in again? If not, maybe it's better to delete them for now?